### DISCUSSION TITLE
x-mask makes form ignore minLength attribute

### DISCUSSION THREAD

**Discussion by Vacate9643** [alpinejs/alpine#4261] [7. Bugs (archived)] [answered]
```html
<form>
  <input type="text" x-mask="999" minlength="3" required />
  <button type="submit">Submit</button>
</form>
```
When I write 2 numbers into the field and click the button, it submits.

Any way to troubleshoot?

**Comment by ekwoka** [comment]
So this happens because programmatically setting the value treats the value as valid.... 🤔

**Reply by Lippiece** [reply]
Do you mean this is an intended behaviour? I would like to use x-mask and have the basic attributes working.

**Reply by ekwoka** [reply, marked as answer]
https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#limiting-user-input-length:-the-maxlength-attribute

Well, it's in the spec, that scripts modifying the value do not trigger validation for min and max length. There isn't a way I can identify to do "force validation" of any kind.

Might not be ideal, but other validations do still apply, so you could do `pattern=".{3,}"` as well...The error will be a pattern failure and not a min length failure though, if you are relying on the native validation messages. So you'd have to cover that.

**Reply by Lippiece** [reply]
Thanks, I didn't think of the pattern attribute. I think this might be the solution. Thank you.

**Reply by SimoTod** [reply]
Probably x-mask should call el.checkValidity to avoid this cases.

**Reply by ekwoka** [reply]
checkValidity still uses the same rules.

so if the last change was not a user input, it will not validate the length. It's all the same internals.

The only thing that mask could do is explicitly check and validate the min and max length itself and set custom validity to imitate the native validity. That feels a bit too involved...

Either way, at least documenting this will go a ways.
